core/blockchain.go

```diff
@@ -253,6 +254,15 @@ func NewBlockChain(db ethdb.Database, cacheConfig *CacheConfig, chainConfig *par
 	}
 	bc.validator = NewBlockValidator(chainConfig, bc, engine)
 	bc.prefetcher = newStatePrefetcher(chainConfig, bc, engine)
+	tp := state.NewTriePrefetcher(bc.stateCache)
+
+	go func() {
+		bc.wg.Add(1)
```

Comment:
Please call `Add` in the outer goroutine instead of the one that's being launched.